test/boost/network_topology_strategy_test.cc

```diff
@@ -634,43 +635,43 @@ SEASTAR_TEST_CASE(test_invalid_dcs) {
 namespace locator {
 
 void topology::test_compare_endpoints(const inet_address& address, const inet_address& a1, const inet_address& a2) const {
-    std::optional<int> expected;
+    auto expected = std::partial_ordering::unordered;
```

Comment:
How can we have unordered here? I understand equivalent/less/greater, but what is unordered here?

Reply:
It expresses that the comparison function is permitted to choose the ordering arbitrarily (meaning that the test doesn't enforce the choice) in some cases (here: when the candidate nodes are both in the same DC, but different racks than the source node).

Note that the comparison function returns `weak_ordering`, not `partial_ordering`, so it's never `unordered`. The test just uses `partial_ordering::unordered` as a "semantically flavoured" version of `optional<weak_ordering>`, where `expected == nullopt` means that we don't care about the result.

Reply:
exactly

Reply:
I can keep it as optional in the test if it makes things any clearer.